### PR TITLE
Modal with Navigation/Tabs

### PR DESCRIPTION
**Pull request by hamishtaplin**
### What is the context of this PR?

[Trello card](https://trello.com/c/qdAqh5Gd/466-validation-modal)

screenshot 2018-07-20 09 19 12

Adds a Modal with tabs navigation for use with validation.

- new Tabs component with Redux implementation—this can be used for any generic tabs implementation although we'd have to look at renaming/moving the existing Tabs component.
- ModalWithNav implement Modal with a Grid layout and Tabs
- content of modal out of scope of this story
- Tabs use [this example](https://www.w3.org/TR/wai-aria-practices/examples/tabs/tabs-1/tabs.html) for aria implementation

### How to review 

- visit storybook entry
- confirm navigation works as expected

### CODE REVIEW

File: src/components/ModalWithNav/index.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+const mapStateToProps = (state, ownProps) => ({
+  activeTabId: state.tabs[ownProps.id]
+});
+
+const mapDispatchToProps = dispatch => ({
+  gotoTab: (tabsId, activeTabId) => dispatch(gotoTab(tabsId, activeTabId))
+});
+
+export default connect(mapStateToProps, mapDispatchToProps)(
```
Comment (SamGodwin2) [inline src/components/ModalWithNav/index.js]: I'm never quite sure how things work with connect but can you do `connect(mapStateToProps, {gotoTab})` and get rid of the map dispatch to props?

File: src/components/ModalWithSidebar/Nav.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  padding: 0;
+`;
+
+export const NavItem = styled.li`
+  margin: 0;
+  padding: 0;
+`;
+
+export const NavBtn = styled.button`
+  --color-text: ${colors.darkGrey};
```
Comment (adammahmood) [inline src/components/ModalWithSidebar/Nav.js]: Nice! Wasn't aware of CSS custom properties/variables with '--' notation. Reference here for others: 
https://drafts.csswg.org/css-variables/

File: src/components/ModalWithNav/index.js (hunk 1 above)
Comment (hamishtaplin) [reply src/components/ModalWithNav/index.js]: ah yeah maybe, will give it a go

**Comment by samiwel** [comment]
[Validation - Modal](https://trello.com/c/qdAqh5Gd/466-validation-modal)

**Pull request merged**